src/hotspot/share/gc/shenandoah/shenandoahVMOperations.cpp

```diff
@@ -61,42 +80,49 @@ void VM_ShenandoahReferenceOperation::doit_epilogue() {
 
 void VM_ShenandoahInitMark::doit() {
   ShenandoahGCPauseMark mark(_gc_id, "Init Mark", SvcGCMarker::CONCURRENT);
+  set_active_generation();
```

Comment:
Why is it here, and not down in `entry_*` methods, probably in helper classes?

Reply:
I think it's here for the same reason `propagate_gc_state_to_java_threads` is here. Having it here makes it easier to see that this critical synchronization step happens for every safepoint.

Reply:
I'm inclined to leave as is, not wanting to disrupt stability of code that is known to be working.  But if there is strong preference otherwise, we can make a change...

Reply:
OK, but file a bug for it to clean up.